packages/plugins/java/kotlin/tests/kotlin.spec.ts

```diff
@@ -0,0 +1,226 @@
+import '@graphql-codegen/testing';
+import { buildSchema } from 'graphql';
+import { plugin } from '../src/index';
+
+const OUTPUT_FILE = 'com/kotlin/generated/resolvers.kt';
+
+describe('Kotlin', () => {
+  // language=GraphQL
+  const schema = buildSchema(`
+    type Query {
+      me: User!
+      user(id: ID!): User!
+      searchUser(searchFields: SearchUser!): [User!]!
+    }
+
+    input InputWithArray {
+      f: [String]
+      g: [SearchUser]
+    }
+
+    input SearchUser {
+      username: String
+      email: String
+      name: String
+      sort: ResultSort
+      metadata: MetadataSearch
+    }
+
+    input MetadataSearch {
+      something: Int
+    }
+
+    enum ResultSort {
+      ASC
+      DESC
+    }
+
+    interface Node {
+      id: ID!
+    }
+
+    type User implements Node {
+      id: ID!
+      username: String!
+      email: String!
+      name: String
+      friends(skip: Int, limit: Int): [User!]!
+      hobbies(skip: Int = 0, limit: Int! = 10): [String!]!
+    }
+
+    type Chat implements Node {
+      id: ID!
+      users: [User!]!
+      title: String
+    }
+
+    enum UserRole {
+      ADMIN
+      USER
+      EDITOR
+    }
+
+    union SearchResult = Chat | User
+  `);
+
+  // TODO need a parser
+  // it('Should produce valid Kotlin code', async () => {
+  //   const result = await plugin(schema, [], {}, { outputFile: OUTPUT_FILE }) as string;
+  //
+  //   validateJava(result);
+  // });
+
+  describe('Config', () => {
+    it('Should use the correct package name by default', async () => {
+      const result = await plugin(schema, [], {}, { outputFile: OUTPUT_FILE });
+
+      expect(result).toContain(`package com.kotlin.generated\n`);
+    });
+
+    it('Should use the package name provided from the config', async () => {
+      const result = await plugin(schema, [], { package: 'com.my.package' }, { outputFile: OUTPUT_FILE });
+
+      expect(result).toContain(`package com.my.package\n`);
+    });
+  });
+
+  describe('Enums', () => {
+    it('Should generate basic enums correctly', async () => {
+      const result = await plugin(schema, [], {}, { outputFile: OUTPUT_FILE });
+
+      // language=kotlin
+      expect(result).toBeSimilarStringTo(`    enum class UserRole(val label: String) {
```

Comment:
If you are basically expect a specific output, you can use Jest snapshot, it will make it easier for you to test.